conf/adcm/python_scripts/run_ldap_sync.py

```diff
@@ -63,27 +75,30 @@ def settings(self):
                 error = AdcmEx(error_code)
                 sys.stdout.write(error.msg)
                 raise error
+            self._settings["DEFAULT_USER_SEARCH"] = _get_user_search(_get_ldap_config())
         return self._settings
 
     def sync(self):
         ldap_groups = self.sync_groups()
-        if ldap_groups:
-            self.sync_users(ldap_groups)
-        else:
-            sys.stdout.write("No groups found. Aborting sync users")
+        self.sync_users(ldap_groups)
 
     def sync_groups(self):
         """Synchronize LDAP groups with group model and delete groups which is not found in LDAP"""
-        self.settings["GROUP_SEARCH"].filterstr = f"(&" \
-                                                  f"(objectClass={self.settings['GROUP_OBJECT_CLASS']})" \
-                                                  f"{self.settings['GROUP_FILTER']})"
-        ldap_groups = self.settings["GROUP_SEARCH"].execute(self.conn, {})
-        self._sync_ldap_groups(ldap_groups)
-        sys.stdout.write("Groups were synchronized\n")
+        ldap_groups = []
+        if self._group_search_configured:
+            self.settings["GROUP_SEARCH"].filterstr = f"(&" \
```

Comment:
Using parentheses allow to avoid backslashes
```
(
    f"{var} text"
    f"text {var}"
    "text"
)

Reply:
done